ecr/resolver_test.go

```diff
@@ -49,7 +50,13 @@ func TestParseImageManifestMediaType(t *testing.T) {
 	} {
 		t.Run(sample.MediaType(), func(t *testing.T) {
 			t.Logf("content: %s", sample.Content())
-			actual := parseImageManifestMediaType(context.Background(), sample.Content())
+			actual, err := parseImageManifestMediaType(context.Background(), sample.Content())
+			if sample == testdata.EmptySample {
+				assert.Error(t, err)
+				assert.True(t, errors.Is(err, ErrInvalidManifest))
+				return
+			}
+			require.NoError(t, err)
```

Comment:
nit: this can be an assert as well - the test cases expects that there's not an error *and* that the mediaType is as expected. There's no "NPE" we might hit on the way, so may as well present all the unmet expectations for a given failure rather than terminating earlier on.